src/pointer-analysis/value_set_analysis.h

```diff
@@ -52,14 +47,50 @@ class value_set_analysis_templatet:
       dest);
   }
 
+  virtual void initialize(const goto_programt &goto_program) override
+  {
+    baset::initialize(goto_program);
+    forall_goto_program_instructions(i_it, goto_program)
+    {
+      static_cast<domaint &>(this->get_state(i_it)).value_set.location_number =
+        i_it->location_number;
+    }
+  }
+
+  void analyze_all_functions(const goto_modelt &goto_model)
+  {
+    // Unlike operator(), which starts at __CPROVER_start and only analyses
+    // reachable code, assume all functions are externally reachable.
+    namespacet ns(goto_model.symbol_table);
+
+    baset::initialize(goto_model.goto_functions);
+
+    forall_goto_functions(it, goto_model.goto_functions)
+    {
+      if(it->second.body.instructions.empty())
+        continue;
+      const auto &start_state =
+        static_cast<const domaint &>(
+          this->find_state(it->second.body.instructions.begin()));
+      if(start_state.is_bottom())
+      {
+        this->entry_state(it->second.body);
```

Comment:
The goal here was probably to preserve existing behaviour. I'm not sure if the current behaviour makes much sense though. If the intended semantics is that the function could be called by an outside caller in any state/context then we should set the state of the entry location to top here (and remove the check `is_bottom()`).

Reply:
I see. I guess the old behaviour was a bit strange -- essentially, if the iteration found the function before a callee did, then we would suppose it to be callable externally and analyse it assuming no knowledge about parameters etc. If the function was found via a callee first then we wouldn't assume this.

This relates to the fact that the analysis doesn't distinguish "may have any value" from "isn't known to have any values (yet)". @tautschnig @kroening what do you prefer -- mimicry of `static_analysist` behaviour, or throw this away completely and essentially ask the user to ensure `__CPROVER_start` exists, or implement @danpoe's suggested distinct "analyse everything" mode?

Reply:
We should definitively have some way of supporting "no `__CPROVER_start` is available." I'd vote for an analyse-everything option/approach.

Reply:
Incidentally this means the "top" state is not really top, it's the entry state (the one above bottom but below all other states)

Reply:
I think we should mimic the existing behaviour for now but look at which use-cases actually need this and longer-term look at removing it.